Packs/CriminalIP/Integrations/CriminalIP/CriminalIP.py

```diff
@@ -0,0 +1,271 @@
+    if summary.get("dga_score", 0) >= 8: body.append(f"DGA score {summary['dga_score']}")
+    if summary.get("newborn_domain"):
+        nd = dateparser.parse(summary["newborn_domain"])
+        if nd and (datetime.datetime.now() - nd).days < 30:
+            body.append(f"Newborn: {summary['newborn_domain']}")
+
+    if not body:
+        return CommandResults("No suspicious element", "CriminalIP.Email_Body", "email_body",
+                              {"domain": domain, "scan_id": scan_id, "body": ""}, raw)
+
+    body_text = f"Domain {domain} scan summary:\n- " + "\n- ".join(body)
+    return CommandResults(body_text, "CriminalIP.Email_Body", "email_body",
+                          {"domain": domain, "scan_id": scan_id, "body": body_text}, raw)
+
+
+def micro_asm(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Create ASM style summary (open ports, CVEs, cert expiry, abuse logs).
+
+    Args:
+        client (Client): API client.
+        args (dict): Command args. Requires 'domain' and 'scan_id'.
+
+    Returns:
+        CommandResults: Summary text.
+    """
+    domain, scan_id = args.get("domain"), args.get("scan_id")
+    raw = client.domain_full_scan_result(scan_id)
+    data = raw.get("data", {})
+
+    results: List[str] = []
+
+    # Certificate expiry
+    for cert in data.get("certificates", []):
+        vto = cert.get("valid_to")
+        dt = dateparser.parse(vto) if vto else None
+        if dt and (dt - datetime.datetime.now()).days < 30:
+            results.append("Certificate expiring soon")
+
+    # Abuse record
+    abuse = (data.get("network_logs") or {}).get("abuse_record", {})
+    if abuse.get("critical", 0) or abuse.get("dangerous", 0):
+        results.append(f"Abuse records: {abuse}")
+
+    # Suspicious logs
+    logs = (data.get("network_logs") or {}).get("data", [])
+    if any(log.get("url", "").endswith(".exe") for log in logs):
+        results.append("Found .exe URL in logs")
+
+    if not results:
+        return CommandResults("No suspicious element", "CriminalIP.Micro_ASM", "micro_asm",
+                              {"domain": domain, "scan_id": scan_id, "result": ""}, raw)
+
+    text = f"===== {domain} =====\n" + "\n".join(results)
+    return CommandResults(text, "CriminalIP.Micro_ASM", "micro_asm",
+                          {"domain": domain, "scan_id": scan_id, "result": text}, raw)
+
+
+# ---------------- Main ----------------
+
+def main() -> None:
+    params = demisto.params()
+    command = demisto.command()
+
+    client = Client(
+        base_url=BASE_URL,
+        verify=not params.get("insecure", False),
+        proxy=params.get("proxy", False),
+        headers={"x-api-key": (params.get("credentials") or {}).get("password")}
+    )
+
+    try:
+        if command == "test-module":
+            try:
+                resp = client.ip_lookup("8.8.8.8")
+                return_results("ok" if resp and "ip" in resp else "Test failed: Unexpected response")
```

Comment:
if it fails the failing error but be informative, for example it's better to add the raw response from the api side to the exception thrown